src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -0,0 +1,118 @@
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Initializing LinuxContainerInitializationService.");
+            _cancellationToken = cancellationToken;
+            await Run();
+        }
+
+        private async Task Run()
+        {
+            if (_settingsManager.IsLinuxContainerEnvironment)
+            {
+                await InitializeAssignmentContext();
+            }
+        }
+
+        private async Task InitializeAssignmentContext()
+        {
+            var startContext = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContext);
+
+            // Container start context is not available directly
+            if (string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("AssignmentContext not available in ContainerStartContext");
+                // Check if the context is available in blob
+                var sasUri = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContextSasUri);
+
+                if (!string.IsNullOrEmpty(sasUri))
+                {
+                    _logger.LogInformation("AssignmentContext ContainerStartContextSasUri available");
+                    startContext = await GetAssignmentContextFromSasUri(sasUri);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("Assigning HostAssignmentContext.");
+
+                var encryptedAssignmentContext = JsonConvert.DeserializeObject<EncryptedHostAssignmentContext>(startContext);
+                var containerKey = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerEncryptionKey);
+                var assignmentContext = encryptedAssignmentContext.Decrypt(containerKey);
+                if (_instanceManager.StartAssignment(assignmentContext))
+                {
+                    _logger.LogInformation("Assign HostAssignmentContext success");
+                }
+                else
+                {
+                    _logger.LogError("Assign HostAssignmentContext failed");
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Waiting for /assign to receive AssignmentContext");
+            }
+        }
+
+        private async Task<string> GetAssignmentContextFromSasUri(string sasUri)
+        {
+            try
+            {
+                return await Read(sasUri);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error calling {nameof(GetAssignmentContextFromSasUri)}");
+            }
+
+            return string.Empty;
+        }
+
+        // virtual for unit testing
+        public virtual async Task<string> Read(string uri)
+        {
+            var cloudBlockBlob = new CloudBlockBlob(new Uri(uri));
+            if (await cloudBlockBlob.ExistsAsync(null, null, _cancellationToken))
```

Comment:
If the blob doesn't exist, you'd swallow that. GetAssignmentContextFromSasUri should error in that case right?

Reply:
Before specialization the blob won't exist (but the url will), its expected and not an error. After specialization the blob will be uploaded and if the container restarts it will read the blob.